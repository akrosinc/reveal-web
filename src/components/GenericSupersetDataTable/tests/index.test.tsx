import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { GenericSupersetDataTable } from '..';

const props = {
  data: [
    ['11 - 15', 9, 6, 3],
    ['< 6', 6, 0, 0],
  ],
  fetchItems: jest.fn(),
  headerItems: ['Age', 'col-1', 'col-2', 'col-3'],
  service: jest.fn(),
  supersetSliceId: '1',
};

describe('components/GenericSupersetDataTable', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    shallow(<GenericSupersetDataTable {...props} />);
  });

  it('should render list correctly', () => {
    const wrapper = mount(<GenericSupersetDataTable {...props} />);

    expect(wrapper.find('ListView').length).toEqual(1);
    expect(toJson(wrapper.find('ListView table'))).toMatchSnapshot('table with data');
    // two reports rendered
    expect(wrapper.find('.listview-tbody tr').length).toEqual(2);
  });

  it('should render headers when there is no data correctly', () => {
    props.data = [];
    const wrapper = mount(<GenericSupersetDataTable {...props} />);

    expect(wrapper.find('.listview-tbody tr').length).toEqual(0);
    expect(wrapper.find('GenericSupersetDataTable div div').text()).toEqual('No rows found');
  });
});
