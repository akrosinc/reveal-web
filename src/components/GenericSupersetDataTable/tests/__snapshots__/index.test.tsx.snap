// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GenericSupersetDataTable should render list correctly: table with data 1`] = `
<table
  className="table table-striped table-bordered plans-list"
>
  <thead
    className="listview-thead"
  >
    <tr
      className="listview-tr"
    >
      <ElementMap
        HTMLTag="th"
        className="listview-td"
        items={
          Array [
            "Age",
            "col-1",
            "col-2",
            "col-3",
          ]
        }
      >
        <th
          className="listview-td"
          key="0"
        >
          Age
        </th>
        <th
          className="listview-td"
          key="1"
        >
          col-1
        </th>
        <th
          className="listview-td"
          key="2"
        >
          col-2
        </th>
        <th
          className="listview-td"
          key="3"
        >
          col-3
        </th>
      </ElementMap>
    </tr>
  </thead>
  <tbody
    className="listview-tbody"
  >
    <tr
      className="listview-tr"
      key="0"
    >
      <ElementMap
        HTMLTag="td"
        className="listview-th"
        items={
          Array [
            "11 - 15",
            9,
            6,
            3,
          ]
        }
      >
        <td
          className="listview-th"
          key="0"
        >
          11 - 15
        </td>
        <td
          className="listview-th"
          key="1"
        >
          9
        </td>
        <td
          className="listview-th"
          key="2"
        >
          6
        </td>
        <td
          className="listview-th"
          key="3"
        >
          3
        </td>
      </ElementMap>
    </tr>
    <tr
      className="listview-tr"
      key="1"
    >
      <ElementMap
        HTMLTag="td"
        className="listview-th"
        items={
          Array [
            "< 6",
            6,
            0,
            0,
          ]
        }
      >
        <td
          className="listview-th"
          key="0"
        >
          &lt; 6
        </td>
        <td
          className="listview-th"
          key="1"
        >
          6
        </td>
        <td
          className="listview-th"
          key="2"
        >
          0
        </td>
        <td
          className="listview-th"
          key="3"
        >
          0
        </td>
      </ElementMap>
    </tr>
  </tbody>
</table>
`;
